src/core/view/background/StavesBackgroundView.cpp

```diff
@@ -0,0 +1,54 @@
+#include "StavesBackgroundView.h"
+
+#include <cmath>
+
+#include "model/BackgroundConfig.h"
+#include "util/Util.h"
+
+using namespace background_config_strings;
+using namespace xoj::view;
+
+StavesBackgroundView::StavesBackgroundView(double pageWidth, double pageHeight, Color backgroundColor,
+                                           const BackgroundConfig& config):
+        OneColorBackgroundView(pageWidth, pageHeight, backgroundColor, config, DEFAULT_LINE_WIDTH, DEFAULT_LINE_COLOR,
+                               ALT_DEFAULT_LINE_COLOR) {}
+
+void StavesBackgroundView::draw(cairo_t* cr) const {
+    // Paint the background color
+    PlainBackgroundView::draw(cr);
+
+    // Get the bounds of the mask, in page coordinates
+    double minX;
+    double maxX;
+    double minY;
+    double maxY;
+    cairo_clip_extents(cr, &minX, &minY, &maxX, &maxY);
+
+    const double staffHeight = lineWidth + 4 * LINES_SPACING;
+
+    // The + 4 * lineWidth is added to preserve the previous behaviour.
```

Comment:
```
const double vOffsetBetweenStaves = STAVES_SPACING + staffHeight + 4 * lineWidth;
```
The point is: this formula makes little to no sense to me: the `+ 4 * lineWidth` was most likely a mistake in the original implementation. We cannot fix this, however, as it would alter how the background is painted in old documents (moving some straves up). Hence my comment. I'll extend the comment to make this more explicit.